src/main/java/pl/sda/refactoring/service/OrderService.java

```diff
@@ -0,0 +1,156 @@
+package pl.sda.refactoring.service;
+
+import lombok.NonNull;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+import pl.sda.refactoring.entity.Currency;
+import pl.sda.refactoring.entity.Order;
+import pl.sda.refactoring.entity.OrderItem;
+import pl.sda.refactoring.entity.OrderStatus;
+
+import java.math.BigDecimal;
+import java.time.Instant;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.UUID;
+
+import static java.math.RoundingMode.HALF_UP;
+import static java.util.UUID.randomUUID;
+
+@Service
+@Transactional
+public class OrderService {
+
+    private final DiscountService discountService;
+    private final CurrencyService currencyService;
+    private final CustomerService customerService;
+    private final OrderRepo orderRepo;
+    private final EmailService emailService;
+    private final List<LocalDate> freeDeliveryDays;
+    private final List<String> emailRecipients;
+
+    public OrderService(@NonNull DiscountService discountService,
```

Comment:
This constructor has 7 dependencies. It may be an indicator of an "God Object". COnsider removing some of its responsibilities to separate objects.